### ISSUE TITLE
levmar tolerance setting

### ISSUE THREAD

**Issue by anetasie** [sherpa/sherpa#785]
Sherpa incorporates the `lmdif`  algorithm which is designed for a minimization of a generic function, so for more general solutions than the minimization problems in Sherpa.

There are three convergence tests in `lmdif`  (i.e. `xtol`, `ftol`,` gtol`).  The main test is based on "an estimate of the distance between the Euclidean norm`||F(x)||` of the residuals at the current approximation `x` and the optimal value `||F(x*)||` at an actual solution `x*` of the problem. It is to guarantee: `|| F(x) || < (1+ftol) || F(x*) ||`
`ftol` recommended setting is at the machine precision. In my tests this tolerance had the most impact on the number of iterations, `nfev`, but still did not significantly impact the final few `delta(chi2)` steps showing higher accuracy than we were planning to get.

Sherpa tolerance setting for the statistics function [mainly `delta(chi2)`] could be at `0.01 (or 0.001)` for most cases, and only for the high S/N data this tolerance would need to be much stricter. We may consider an addition of a new tolerance parameter to `levmar` that would allow for testing directly the difference in the statistic values. This may need to be incorporated into `lmdif`. 
However, we still need to run tests to understand whether this can significantly improve the performance.

**Comment by dtnguyen2** [comment]
Regarding the statement:

>  We may consider an addition of a new tolerance parameter to levmar that would allow for 
>  testing directly the difference in the statistic values.

Isn't that what It is the following condition:  ```|| F(x) || < (1+ftol) || F(x*) ||``` is?

**Comment by anetasie** [comment]
@dtnguyen2 yes, this is the condition which is embedded deeply in the `lmdif`. You know that there are three conditions for the convergence with `xtol` and `ftol` linked. Increasing the`xtol` which is directly related to the jacobian often results in increased number of `fev`. `ftol` increase reduces the `nfev`, but as you can see this is not a simple relation with delta(chi2).
